tests/test_pipelines/test_auto_augment.py

```diff
@@ -0,0 +1,107 @@
+
+    # test assertion for invalid value of prob
+    with pytest.raises(AssertionError):
+        transform = dict(type='Shear', magnitude=0.5, prob=100)
+        build_from_cfg(transform, PIPELINES)
+
+    # test assertion for invalid direction
+    with pytest.raises(AssertionError):
+        transform = dict(type='Shear', magnitude=0.5, direction='diagonal')
+        build_from_cfg(transform, PIPELINES)
+
+    # test assertion for invalid value of random_negative_prob
+    with pytest.raises(AssertionError):
+        transform = dict(type='Shear', magnitude=0.5, random_negative_prob=100)
+        build_from_cfg(transform, PIPELINES)
+
+    # test case when magnitude = 0, therefore no shear
+    results = construct_toy_data()
+    transform = dict(type='Shear', magnitude=0., prob=1.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    assert (results['img'] == results['ori_img']).all()
+
+    # test case when prob = 0, therefore no shear
+    results = construct_toy_data()
+    transform = dict(type='Shear', magnitude=0.5, prob=0.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    assert (results['img'] == results['ori_img']).all()
+
+    # test shear horizontally, magnitude=1
+    results = construct_toy_data()
+    transform = dict(
+        type='Shear', magnitude=1, pad_val=0, prob=1., random_negative_prob=0.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    sheared_img = np.array([[1, 2, 3, 4], [0, 5, 6, 7], [0, 0, 9, 10]],
+                           dtype=np.uint8)
+    sheared_img = np.stack([sheared_img, sheared_img, sheared_img], axis=-1)
+    assert (results['img'] == sheared_img).all()
+    assert (results['img'] == results['img2']).all()
+
+    # test shear vertically, magnitude=-1
+    results = construct_toy_data()
+    transform = dict(
+        type='Shear',
+        magnitude=-1,
+        pad_val=0,
+        prob=1.,
+        direction='vertical',
+        random_negative_prob=0.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    sheared_img = np.array([[1, 6, 11, 0], [5, 10, 0, 0], [9, 0, 0, 0]],
+                           dtype=np.uint8)
+    sheared_img = np.stack([sheared_img, sheared_img, sheared_img], axis=-1)
+    assert (results['img'] == sheared_img).all()
+
+    # test shear vertically, magnitude=1, random_negative_prob=1
+    results = construct_toy_data()
+    transform = dict(
+        type='Shear',
+        magnitude=1,
+        pad_val=0,
+        prob=1.,
+        direction='vertical',
+        random_negative_prob=1.)
+    pipeline = build_from_cfg(transform, PIPELINES)
+    results = pipeline(results)
+    sheared_img = np.array([[1, 6, 11, 0], [5, 10, 0, 0], [9, 0, 0, 0]],
+                           dtype=np.uint8)
+    sheared_img = np.stack([sheared_img, sheared_img, sheared_img], axis=-1)
+    assert (results['img'] == sheared_img).all()
```

Comment:
we may add more test cases, e.g., test regarding pad_val. 
https://github.com/open-mmlab/mmcv/blob/master/tests/test_image/test_geometric.py#L447